src/Shared/ServerInfrastructure/StringUtilities.cs

```diff
@@ -418,6 +529,36 @@ private static bool CheckBytesInAsciiRange(Vector<sbyte> check)
             return Vector.GreaterThanAll(check, Vector<sbyte>.Zero);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool CheckBytesInAsciiRange(Vector256<sbyte> check, Vector256<sbyte> zero)
```

Comment:
Out of curiosity, why is the zero vector passed in as an argument in these methods?

Reply:
So that the "zero register" can be reused. Otherwise the JIT won't reuse that register and xors a register again. 
Although zeroing (xor) a register has no latency, the cpu-frontend must fetch and decode the instruction, so it still has some cost, that can be avoided with this way.
Note: this works only because this method will be inlined.